### PR TITLE
WIP: factorizing tight bbox calculation and application

### PR DESCRIPTION
**Pull request by choldgraf**
See #7226 for some discussion. This creates a `tight_bbox` method so it can be called without necessarily needing to save the figure to disk.

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -2241,6 +2191,68 @@ def print_figure(self, filename, dpi=None, facecolor=None, edgecolor=None,
             #self.figure.canvas.draw() ## seems superfluous
         return result
 
+    def tight_bbox(self, canvas, dpi, facecolor, edgecolor,
```
Comment (NelleV) [inline lib/matplotlib/backend_bases.py]: Does this need to be a public function?

Comment (choldgraf) [inline lib/matplotlib/backend_bases.py]: (this is the PR I was looking for earlier). I think the current push is a pretty rough attempt at the `tight_bbox` method. It should be public, and probably ideally a method of `Figure`. This doesn't accomplish that, correct? I'm hesitant to include too many changes in this PR because this code is quite messy and a bit hard to follow

**Comment by choldgraf** [comment]
I'm going to close this because it seems like this isn't worth the hassle that it'd take to spin off a new method...for my use case, it's probably better to change the way I'm creating figures / axes...
